Src/AutoFixture/ElementsBuilder.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Ploeh.AutoFixture.Kernel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Draws a random element from the given collection
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public sealed class ElementsBuilder<T> : ISpecimenBuilder
+    {
+        private readonly ICollection<T> elements;
+        private readonly RandomNumericSequenceGenerator sequence;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ElementsBuilder{T}"/> class.
+        /// </summary>
+        /// <param name="elements">The collection from which elements should be drawn from.
+        /// It must contain at least two element.</param>
+        public ElementsBuilder(ICollection<T> elements)
+        {
+            if (elements == null)
+                throw new ArgumentNullException("elements");
+
+            if (elements.Count < 2)
```

Comment:
I think that singleton collections should be possible as well, even if that requires a special case in the implementation. That is, after all, an implementation detail.

Whether or not each element should be returned exactly once before restarting the 'pool' is not something I feel strongly about. It *would* be in the spirit of how AutoFixture currently tends to do things, so I'd slightly lean towards that behaviour for the sake of consistency.